wapitiCore/attack/mod_http_post.py

```diff
@@ -0,0 +1,46 @@
+from httpx import RequestError
+
+from wapitiCore.attack.attack import Attack
+from wapitiCore.net.web import Request
+from wapitiCore.language.vulnerability import MEDIUM_LEVEL, _
+from wapitiCore.definitions.http_post import NAME
+
+
+# This module check the security of transported credentials of login forms
+class mod_http_post(Attack):
+    """Check if credentials are transported on an encrypted channel."""
+    name = "http_post"
+
+    async def must_attack(self, request: Request):
+        # We leverage the fact that the crawler will fill password entries with a known placeholder
+        if "Letm3in_" not in request.encoded_data + request.encoded_params:
+            return False
+
+        # We may want to remove this but if not available fallback to target URL
+        if not request.referer:
+            return False
+
+        return True
+
+    async def attack(self, request: Request):
+        try:
+            page = await self.crawler.async_send(request, follow_redirects=True)
```

Comment:
I don't see any benefit into sending the request again.
Especially the form may be at another URL than the target of the form.

Reply:
You will make things easier and more effective just by checking requests with clear protocol and presence of "Letm3in_".